packages/squiggle-lang/__tests__/Distributions/AlgebraicCombination_test.res

```diff
@@ -0,0 +1,313 @@
+  triangularDist,  // low=1; medium=2; high=3;
+  exponentialDist,  // rate=2
+} = module(GenericDist_Fixtures)
+
+let {
+    algebraicAdd, 
+    algebraicMultiply, 
+    algebraicDivide, 
+    algebraicSubtract, 
+    algebraicLogarithm, 
+    algebraicPower
+} = module(DistributionOperation.Constructors)
+
+let algebraicAdd = algebraicAdd(~env)
+let algebraicMultiply = algebraicMultiply(~env)
+let algebraicDivide = algebraicDivide(~env)
+let algebraicSubtract = algebraicSubtract(~env)
+let algebraicLogarithm = algebraicLogarithm(~env)
+let algebraicPower = algebraicPower(~env)
+
+describe("(Algebraic) addition of distributions", () => {
+
+    describe("mean", () => {
+        test("normal(mean=5) + normal(mean=20)", () => {
+            normalDist5
+            -> algebraicAdd(normalDist20)
+            -> E.R2.fmap(GenericDist_Types.Constructors.UsingDists.mean)
+            -> E.R2.fmap(run)
+            -> E.R2.fmap(toFloat)
+            -> E.R.toExn
+            -> expect
+            -> toBe(Some(2.5e1))
+        })
+
+        test("uniform(low=9, high=10) + beta(alpha=2, beta=5)", () => {
+            // let uniformMean = (9.0 +. 10.0) /. 2.0
+            // let betaMean = 1.0 /. (1.0 +. 5.0 /. 2.0)
+            let received = uniformDist 
+                -> algebraicAdd(betaDist)
+                -> E.R2.fmap(GenericDist_Types.Constructors.UsingDists.mean)
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toExn
+            switch received {
+                | None => "algebraicAdd has" -> expect -> toBe("failed")
+                // This is nondeterministic, we could be in a situation where ci fails but you click rerun and it passes, which is bad. 
+                // sometimes it works with ~digits=2. 
+                | Some(x) => x -> expect -> toBeSoCloseTo(0.01927225696028752, ~digits=1)  // (uniformMean +. betaMean)
+            }
+        })
+        test("beta(alpha=2, beta=5) + uniform(low=9, high=10)", () => {
+            // let uniformMean = (9.0 +. 10.0) /. 2.0
+            // let betaMean = 1.0 /. (1.0 +. 5.0 /. 2.0)
+            let received = betaDist 
+                -> algebraicAdd(uniformDist)
+                -> E.R2.fmap(GenericDist_Types.Constructors.UsingDists.mean)
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toExn
+            switch received {
+                | None => "algebraicAdd has" -> expect -> toBe("failed")
+                // This is nondeterministic, we could be in a situation where ci fails but you click rerun and it passes, which is bad. 
+                // sometimes it works with ~digits=2. 
+                | Some(x) => x -> expect -> toBeSoCloseTo(0.019275414920485248, ~digits=1)  // (uniformMean +. betaMean)
+            }
+        })
+    })
+    describe("pdf", () => {
+        testAll("(normal(mean=5) + normal(mean=5)).pdf (imprecise)", list{8e0, 1e1, 1.2e1, 1.4e1}, x => {
```

Comment:
this test is completely wrong, Nuño thought it was smelly and it took us a couple minutes to realize that the standard deviation in the `normalDist10` fixture is wrong.